engine/src/main/java/org/terasology/entitySystem/systems/RenderSystem.java

```diff
@@ -16,16 +16,32 @@
 package org.terasology.entitySystem.systems;
 
 /**
+ * Interface for component systems that needs to be updated every time certain game elements are rendered (generally to render something).
  */
 public interface RenderSystem extends ComponentSystem {
 
+    /**
+     * Called when opaque objects should be rendered.
```

Comment:
I'd write:

_Called with an opengl state useful to the rendering of opaque objects. See OpaqueObjectsNode for more information._

I'd describe the other methods in a similar fashion.